packages/story-editor/src/components/panels/shared/media/fileRow.js

```diff
@@ -0,0 +1,183 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import styled, { css } from 'styled-components';
+import PropTypes from 'prop-types';
+import {
+  Button,
+  BUTTON_SIZES,
+  BUTTON_TYPES,
+  BUTTON_VARIANTS,
+  Icons,
+  Text,
+  THEME_CONSTANTS,
+  themeHelpers,
+} from '@googleforcreators/design-system';
+import { useState, useCallback } from '@googleforcreators/react';
+import { __ } from '@googleforcreators/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { Row } from '../../../form';
+import Tooltip from '../../../tooltip';
+import DropDownMenu from './dropDownMenu';
+
+const InputRow = styled.div`
+  display: flex;
+  flex-grow: 1;
+  margin: 0 8px;
+  text-overflow: ellipsis;
+`;
+
+const ReplaceIcon = styled(Icons.Rotate)`
+  vertical-align: bottom;
+`;
+
+const LinkIcon = styled(Icons.Link)`
+  width: 24px;
+  height: 24px;
+  vertical-align: bottom;
+`;
+
+const UploadIcon = styled(Icons.ArrowOutline)`
+  width: 24px;
+  height: 24px;
+  vertical-align: bottom;
+`;
+
+const Link = styled.a`
+  ${({ theme }) => css`
+    color: ${theme.colors.fg.primary};
+    text-decoration: none;
+    cursor: pointer;
+
+    :hover,
+    :active,
+    :focus {
+      text-decoration: underline;
+      /* Override WordPress's common css */
+      color: ${theme.colors.fg.primary} !important;
+    }
+
+    ${themeHelpers.focusableOutlineCSS(theme.colors.border.focus)}
+  `};
+`;
+
+const StyledButton = styled(Button)`
+  ${({ theme }) =>
+    themeHelpers.focusableOutlineCSS(
+      theme.colors.border.focus,
+      theme.colors.bg.secondary
+    )};
+`;
+
+const FileName = styled(Text).attrs({
+  size: THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL,
+})`
+  color: ${({ theme }) => theme.colors.fg.primary};
+`;
+
+function FileRow({
+  id,
+  isExternal,
+  src,
+  title,
+  options,
+  removeItemLabel,
+  onRemove,
+  children,
+}) {
+  const optionsWithGroup = [
+    {
+      group: options,
+    },
+  ];
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+
+  const onMenuOpen = () => setIsMenuOpen(true);
+  const onMenuClose = () => setIsMenuOpen(false);
+  const onMenuSelected = useCallback(
+    (value) => {
+      setIsMenuOpen(false);
+      options.find((item) => item.value === value)?.onClick();
+    },
+    [options]
+  );
+
+  return (
+    <Row key={`row-filename-${id}`}>
+      <Tooltip
+        title={
+          isExternal
+            ? __('External file', 'web-stories')
+            : __('Local file', 'web-stories')
+        }
+      >
+        {isExternal ? <LinkIcon /> : <UploadIcon />}
+      </Tooltip>
+      <InputRow>
+        <FileName>
+          <Link href={src} rel="noreferrer" target="_blank">
```

Comment:
Might be nice to add an aria label with for  "External link" + title name here if the file is external

Reply:
Hmm we don't usually prefix external links like that elsewhere in the app. And whether it's an internal or external file is announced with the icon before the link.